src/transactions/OfferExchange.cpp

```diff
@@ -1281,7 +1288,95 @@ exchangeWithPool(int64_t reservesToPool, int64_t maxSendToPool, int64_t& toPool,
     }
 }
 
-ConvertResult
+static PoolID
+getPoolID(Asset const& x, Asset const& y, int32_t feeBps)
+{
+    LiquidityPoolParameters lpp(LIQUIDITY_POOL_CONSTANT_PRODUCT);
+    auto& cp = lpp.constantProduct();
+    cp.assetA = std::min(x, y);
+    cp.assetB = std::max(x, y);
+    cp.fee = feeBps;
+    return xdrSha256(lpp);
+}
+
+static bool
+exchangeWithPool(AbstractLedgerTxn& ltxOuter, Asset const& toPoolAsset,
+                 int64_t maxSendToPool, int64_t& toPool,
+                 Asset const& fromPoolAsset, int64_t maxReceiveFromPool,
+                 int64_t& fromPool, RoundingType round,
+                 int64_t maxOffersToCross)
+{
+    LedgerTxn ltx(ltxOuter); // Always rolls back
```

Comment:
This comment isn't accurate. `ltx` can be committed below.

Reply:
Fixed.